modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/transaction/TransactionalSourceTestCase.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.transaction;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import org.junit.After;
+import org.junit.Test;
+import org.mule.functional.junit4.ExtensionFunctionalTestCase;
+import org.mule.runtime.api.tx.TransactionException;
+import org.mule.runtime.core.api.construct.Flow;
+import org.mule.tck.probe.JUnitLambdaProbe;
+import org.mule.tck.probe.PollingProber;
+import org.mule.test.transactional.TransactionalExtension;
+import org.mule.test.transactional.TransactionalSource;
+import org.mule.test.transactional.connection.MessageStorage;
+import org.mule.test.transactional.connection.TestTransactionalConnection;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.BooleanSupplier;
+
+public class TransactionalSourceTestCase extends ExtensionFunctionalTestCase {
+
+  @Override
+  protected Class<?>[] getAnnotatedExtensionClasses() {
+    return new Class<?>[] {TransactionalExtension.class};
+  }
+
+  @Override
+  protected String getConfigFile() {
+    return "source-transaction-config.xml";
+  }
+
+  @After
+  public void tearDown() {
+    MessageStorage.messages = new ConcurrentLinkedQueue<>();
```

Comment:
MessageStore should have a reset() method. You need to do this on the doSetup() as well